### PR TITLE
Reuse next_subscription filter for is_renewed

### PR DESCRIPTION
**Pull request by kaapstorm**
Context [FB 221066](http://manage.dimagi.com/default.asp?221066)

subscription.is_renewed was returning True but subscription.next_subscription was returning None. Reusing the same filter resolves.

Added tearDown to the test to ensure order of tests doesn't affect results.

@calellowitz @benrudolph cc @dannyroberts @biyeun

### CODE REVIEW

File: corehq/apps/accounting/models.py [hunk 1]
```diff
@@ -1081,25 +1081,27 @@ def allowed_attr_changes(self):
         return ['do_not_invoice', 'no_invoice_reason', 'salesforce_contract_id']
 
     @property
+    def next_subscription_filter(self):
+        return Subscription.objects.filter(
+                subscriber=self.subscriber, date_start__gt=self.date_start
```
Comment (dimagimon) [inline corehq/apps/accounting/models.py]: E126 continuation line over-indented for hanging indent

File: corehq/apps/accounting/models.py [hunk 2]
```diff
@@ -1081,25 +1081,27 @@ def allowed_attr_changes(self):
...
+            ).exclude(
```
Comment (dimagimon) [inline corehq/apps/accounting/models.py]: E121 continuation line under-indented for hanging indent

File: corehq/apps/accounting/models.py [hunk 3]
```diff
@@ -1081,25 +1081,27 @@ def allowed_attr_changes(self):
...
+                pk=self.pk
+            ).exclude(
```
Comment (dimagimon) [inline corehq/apps/accounting/models.py]: E123 closing bracket does not match indentation of opening bracket's line

File: corehq/apps/accounting/models.py [hunk 4]
```diff
@@ -1081,25 +1081,27 @@ def allowed_attr_changes(self):
...
+                date_start=F('date_end')
+            )
```
Comment (dimagimon) [inline corehq/apps/accounting/models.py]: E123 closing bracket does not match indentation of opening bracket's line

**Comment by kaapstorm** [comment]
:white_check_mark:

**Pull request merged**